### PR TITLE
✨Add complications

### LINKED ISSUE #101: Build scaffold for complications
We still need designs for at least a couple of complications, but let's start building what's necessary for them to work.

Take a look at: https://developer.apple.com/documentation/clockkit/adding_a_complication_to_your_watchos_app

We can start with Modular Lage, which has space for a lot of information and it's compatible with older watches. Let's try to include:
- Current timer if there's one running (description, project, time...)
- Total time tracked today if there's no entry running

### PR DESCRIPTION
**Pull request by sofokli1**
Closes #101 

Complications are divided into complications families and each family has sub options. So for each complication family we can show 1 of the sub options. A good list with visual representations can be found in the links below:
https://theswiftdev.com/2016/04/28/clockkit-complications-cheat-sheet/
https://blog.drkaka.com/apple-watch-complication-templates-f78b75a80bb1

- In most cases we have options to show one image and one text (in fewer cases we can show multi line text as well). There are also options for only image and only text.
- The positions of the text and image is fixed and decided automatically by the Template that we chose to use. (sometimes image is in front of text, sometime the other way around, some times image above the text)
- The font is also not up to us and its size may change based on the amount of text shown.
- We can set a tint color (text and/or image) but based on the watchface shown and its customisation from the user, it may be disregarded.
- In some of the complication families the images are masked with white or the color of the whatchface theme (so a round red toggl logo with white in the middle, will only look like a big white circle. in this situations we need a toggl logo where the inner details are cropped, so that regardless of the color of the mask it will still look like the toggl logo)
- Many of the complications provide a gauge of some sort (circular, open ends, or just a curved line) I am not using them but it is up to as to decide if we want to use them. I can see the use of them in a count down timer but not sure in our case.

If we do not have a running time entry, I show either the toggl logo or the logo + name.
If we have a running TE, I show the timer, description where possible and logo if possible (but this needs designs)
When the user customises the watchface complications and sees our app, the complications will look as if it is running a TE.
It updates the complications whenever the app opens and closes. (this should also happen when we get push notifications)

### CODE REVIEW

**Comment by Odrakir** [comment]
I've added the classes I'm using in case it's useful, @sofokli1. This is still pending a code review which I'll do next week.

**Note for designers**: We don't have total freedom to do what we want inside each one of these, still there's some things that we can change. Some texts accept different colors, some don't, I used them to put the project color where appropriate. We can also swap texts and remove them. The Toggl logos can be replaced by something else (and also some can be made bigger, I didn't do it in these screenshots)

All complications show 2 state: Timer stopped and Timer running. I was thinking we could also use the stop state to show some summary of the day (at least for big enough complications), but I didn't do any examples of that.

The last 3 are all circular and very similar. In the Circular Small one I tried different things so you see the possibilities with that one. We have to chose though.

## Graphic Bezel (Infograph watch faces)
```
+ CLKComplicationTemplateGraphicBezelCircularText
- CLKComplicationTemplateGraphicCircularStackImage
- CLKComplicationTemplateGraphicCircularImage
```
bezel

## Graphic Rectangular (Infograph Modular watch faces)
```
- CLKComplicationTemplateGraphicRectangularStandardBody
```
rectangular

## Graphic Corner (Infograph watch faces)
```
- CLKComplicationTemplateGraphicCornerStackText
- CLKComplicationTemplateGraphicCornerTextImage
```
corner

## Utilitarian Small (Motion)
```
- CLKComplicationTemplateUtilitarianSmallFlat
```

utilitarian_s

## Utilitarian Large (Utility, Motion)
```
- CLKComplicationTemplateUtilitarianSmallFlat
```

Utilitaria_l

## Modular Large (Modular watch face)
```
- CLKComplicationTemplateModularLargeStandardBody
```

Modular_l

## Modular Small (Modular watch face)
```
- CLKComplicationTemplateModularSmallSimpleImage
- CLKComplicationTemplateModularSmallStackText
```
modular_s

## Graphic Circular (Infograph and Infograph Modular watch faces)
```
- CLKComplicationTemplateGraphicCircularStackText
- CLKComplicationTemplateGraphicCircularImage
```
GraphicCircular

## Circular Small (Color watch face)
```
- CLKComplicationTemplateCircularSmallStackImage
- CLKComplicationTemplateCircularSmallSimpleText
- CLKComplicationTemplateCircularSmallStackText
- CLKComplicationTemplateCircularSmallSimpleImage
```
CircularSmall

## Extra Large (X-Large watch faces)
```
- CLKComplicationTemplateExtraLargeSimpleImage
- CLKComplicationTemplateExtraLargeStackText
```
extralarge

**Comment by sofokli1** [comment]
Helpful link to visualise possible complications https://theswiftdev.com/2016/04/28/clockkit-complications-cheat-sheet/

**Comment by maritoggl** [comment]
The Graphic Bezel looks a bit weird with the running time entry when the top and bottom of our logo are cut off. Is there a way around that?
Also, the Graphic Corner version - is there a way to rotate the logo so that it would be under the same angle as the text is? 
Other than that, the options proposed seem decent enough to me.

**Comment by AlicjaSuska** [comment]
@Odrakir thank you fo sharing this. For the first case mentioned by @maritoggl (The Graphic Bezel) - how about instead of logo+time, we display time on the red backgound only? It will incorporate our branding and provide good readability at the same time.

Also, in the last example - can we make the logo bigger?

**Comment by sofokli1** [comment]
@maritoggl 
> The Graphic Bezel looks a bit weird with the running time entry when the top and bottom of our logo are cut off. Is there a way around that?

That will be fixed when I get a new SVG image of the toggl logo so that I can size it accordingly. Most likely the logo is going to be smaller since there isn't much space.

> Also, the Graphic Corner version - is there a way to rotate the logo so that it would be under the same angle as the text is?

Couldn't find a way to achieve this. Even apple's own app images in that similar situation are not rotate. Only text is automatically rotated.

**Comment by sofokli1** [comment]
@AlicjaSuska 
> For the first case mentioned by @maritoggl (The Graphic Bezel) - how about instead of logo+time, we display time on the red backgound only? It will incorporate our branding and provide good readability at the same time.

We can not set a color for the background if we are showing text. We can do 
- text only
- image only
- small image on top of text (as our example)
- text on top of text (we can just say Toggl in the top and the timer below it)
Screenshot 2020-02-14 at 4 36 53 PM

- different combinations of open and close gouges + text or image
Screenshot 2020-02-14 at 4 37 21 PM


> Also, in the last example - can we make the logo bigger?

yes we can. I will need an SVG image of the toggl logo with a hollowed out symbol and one with the white symbol. (the one used in this PR is a bit pixelated since I had to make it from some other image)

**Comment by AlicjaSuska** [comment]
@sofokli1 text + time seems reasonable to me. We would avoid the cut logo

The hollowed svg and the white symbol separately
[Logo.zip](https://github.com/toggl/watchos-app/files/4209649/Logo.zip)
:

**Comment by Odrakir** [comment]
Some comments:
- Graphic rectangular dones't seem to be working
- Modular large is not showing the project

Other than that I'm ok with it.

File: TogglTrack/Core/Actions/AppAction.swift [hunk 1]
```diff
@@ -19,6 +19,7 @@ public enum AppAction
     case timeline(TimelineAction)
     case user(LoginAction)
     case loadAll(force: Bool)
+    case loadAllForBackgroundUpdate(completion: ()->())
```
Comment (Odrakir) [inline TogglTrack/Core/Actions/AppAction.swift]: I'm not sure if this is a good idea, I don't really like it, I think feedback should come from subscribing to the state and not from this kind of closure. Let's leave it here, but try not to do this in the main app.

**Pull request merged**